contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -430,7 +430,7 @@ func (q *queuer) Run(done <-chan struct{}) {
 			}
 
 			pod := p.(*Pod)
-			if pod.Spec.NodeName != "" {
+			if _, ok := pod.ObjectMeta.Annotations[annotation.BindingHostKey]; ok {
```

Comment:
Because in upstream the kubelet will notice any task with a pre-set NodeName and launch it. In the k8sm case we have to create a Mesos task, launch an executor and then pass the pod to the kubelet. In other words: if we only filter for NodeName, we miss those pre-scheduled tasks in the k8sm scheduler.